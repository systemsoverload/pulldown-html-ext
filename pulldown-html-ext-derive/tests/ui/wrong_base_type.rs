use pulldown_html_ext_derive::html_writer;

struct HtmlWriterBase<W>(W);

#[html_writer]
struct WrongBaseType<W> {  // This should fail
    base: String,  // Wrong type
}

fn main() {}
