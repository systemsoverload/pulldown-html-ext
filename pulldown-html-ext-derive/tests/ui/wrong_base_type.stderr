error: base field must be of type HtmlWriterBase<W>
 --> tests/ui/wrong_base_type.rs:7:11
  |
7 |     base: String,  // Wrong type
  |           ^^^^^^
